### PR TITLE
update docs for gpu external memory

### PR DESCRIPTION
**Pull request by rongou**
Last bit for #4357 

@RAMitchell @trivialfis

### CODE REVIEW

**Comment by mli** [comment]
# [Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=h1) Report
> Merging [#5332](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=desc) into [master](https://codecov.io/gh/dmlc/xgboost/commit/8aa8ef10317cfc6d625050d98a5b86d3b1b888d3&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master    #5332   +/-   ##
=======================================
  Coverage   83.76%   83.76%           
=======================================
  Files          11       11           
  Lines        2409     2409           
=======================================
  Hits         2018     2018           
  Misses        391      391           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=footer). Last update [8aa8ef1...d98d145](https://codecov.io/gh/dmlc/xgboost/pull/5332?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: doc/parameter.rst [hunk 1]
```diff
@@ -88,6 +88,16 @@ Parameters for Tree Booster
...
 
+* ``sampling_method`` [default= ``uniform``]
+
+  - The method to use to sample the training instances.
+  - ``uniform``: each training instance has an equal probability of being selected. Typically set
+    ``subsample`` >= 0.5 for good results.
+  - ``gradient_based``: the selection probability for each training instance is proportional to the
+    *regularized absolute value* of gradients (more specifically, :math:`\sqrt{g^2+\lambda h^2}`).
+    ``subsample`` may be set to as low as 0.1 without loss of model accuracy. Note that this
+    sampling method is only supported when ``tree_method`` is set to ``gpu_hist``.
```
[the remark this answers is not in the document]
Comment (rongou) [reply doc/parameter.rst]: Done.

File: doc/tutorials/external_memory.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
-############################################
-Using XGBoost External Memory Version (beta)
```
Comment (trivialfis) [inline doc/tutorials/external_memory.rst]: Em ... the external memory support is not available for CPU Hist, if we were to declare it stable, we should at least document it in the first section that `Hist` is not supported.

**Comment by trivialfis** [comment]
Otherwise looks good to me.

File: doc/tutorials/external_memory.rst (hunk 2 above)
Comment (rongou) [reply doc/tutorials/external_memory.rst]: Are you sure? I think I've tried `hist` for comparison in external memory mode. The limitation is right now the CPU code keeps the whole histogram in memory, which may still cause out-of-memory errors if the dataset is too large, but the mechanism of using `#cacheprefix` and dealing with multiple pages is supported.

Comment (trivialfis) [reply doc/tutorials/external_memory.rst]: I'm not.  But there's https://github.com/dmlc/xgboost/issues/4093

Comment (rongou) [reply doc/tutorials/external_memory.rst]: Ok, added `hist` not well tested as a limitation. If you want, I can add the `beta` back. :)

**Pull request merged**